packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts

```diff
@@ -361,6 +386,25 @@ export class TaskDefinition extends TaskDefinitionBase {
    * Adds a new container to the task definition.
    */
   public addContainer(id: string, props: ContainerDefinitionOptions) {
+    if (this.cpu !== undefined
+          && props.cpu !== undefined
+          && parseInt(this.cpu, 10) < props.cpu) {
+      this.node.addWarning('CPU specified for the container cannot be greater than the CPU for the task definition');
+    }
+    if (this.memoryMiB !== undefined
+          && props.memoryLimitMiB !== undefined
+          && parseInt(this.memoryMiB, 10) < props.memoryLimitMiB) {
+      this.node.addWarning('Memory specified for the container cannot be greater than the memory for the task definition');
+    }
+    if (this.remainingMemoryMiB !== undefined
+          && props.memoryLimitMiB !== undefined
+          && parseInt(this.remainingMemoryMiB, 10) > (parseInt(this.remainingMemoryMiB, 10) - props.memoryLimitMiB)) {
+      this.node.addWarning('Total memory specified for all containers cannot be greater than the memory for the task definition');
+      this.remainingMemoryMiB = (parseInt(this.remainingMemoryMiB, 10) - props.memoryLimitMiB).toString();
```

Comment:
The calculation is done in both branches. Deduplicate.

Suggestion to refactor: do subtraction first, add error if `remaining < 0`